From b5c32c77d2415926ef80a70251ac9cfab35f8e6b Mon Sep 17 00:00:00 2001
From: Nell <xianbin.zhu@spacemit.com>
Date: Fri, 21 Jun 2024 15:42:47 +0800
Subject: [PATCH 0777/1233] fix-regulator:do not load the driver asynchronously
 for platform spacemit

if this driver load asynchronously, the drivers that rely on it to
load in advance may not load successully

Change-Id: I0a716d4eceb8f4f729c8f059d1d386921f90ace5
---
 drivers/regulator/fixed.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/regulator/fixed.c b/drivers/regulator/fixed.c
index 254c0a8a45559..7f7f12d33d082 100644
--- a/drivers/regulator/fixed.c
+++ b/drivers/regulator/fixed.c
@@ -383,7 +383,9 @@ static struct platform_driver regulator_fixed_voltage_driver = {
 	.probe		= reg_fixed_voltage_probe,
 	.driver		= {
 		.name		= "reg-fixed-voltage",
+#ifndef CONFIG_SOC_SPACEMIT
 		.probe_type	= PROBE_PREFER_ASYNCHRONOUS,
+#endif
 		.of_match_table = of_match_ptr(fixed_of_match),
 	},
 };
-- 
2.52.0

